### PR TITLE
Compute only unique minterms during mintermization

### PR DESCRIPTION
**Pull request by jurajsic**
This PR should improve the speed of mintermization, by computing minterms only for unique BDDs from transitions + only keeping unique minterms.

### CODE REVIEW

File: src/mintermization.cc [hunk 1]
```diff
@@ -115,37 +115,30 @@ void Mata::Mintermization::trans_to_bdd_afa(const IntermediateAut &aut)
...
     }
 }
 
-std::vector<BDD> Mata::Mintermization::compute_minterms(const std::vector<BDD>& bdds)
+std::unordered_set<BDD> Mata::Mintermization::compute_minterms(const std::unordered_set<BDD>& bdds)
 {
-    std::vector<BDD> stack;
-    // we start mintermization with first bdd and its negation
-    stack.push_back(bdds.front());
-    stack.push_back(!bdds.front());
```
Comment (tfiedor) [inline src/mintermization.cc:128]: What about the mentioned negation? Shouldn't it be added as well? Or is it included in the `bdd_mng.bddOne()`?

**Review by tfiedor — APPROVED** [review]
Seems OK

**Review by Adda0 — APPROVED** [review]
The changes look good to me. We will fix the Python binding once the PR is ready to be merged.

File: src/mintermization.cc (hunk 1 above)
Comment (jurajsic) [reply src/mintermization.cc:128]: BDD at front is now processed in the following loop where we do both `1 && front` and `1 && !front`, so we add it to the set of minterms

**Comment by jurajsic** [comment]
Is there mintermization in python binding? I changed the interface a bit, so it could break it there. I also changed one header file in cudd, I am not sure whether it should be also changed in `cudd-min` for binding.

**Comment by tfiedor** [comment]
> Is there mintermization in python binding? I changed the interface a bit, so it could break it there. I also changed one header file in cudd, I am not sure whether it should be also changed in `cudd-min` for binding.

Should I fix the binding? You have nothing more to commit?

**Comment by tfiedor** [comment]
The binding should be fixxed.

**Comment by Adda0** [comment]
I am fine with the changes. Do any of you plan to add anything new to the PR, or can we merge the PR?

**Comment by tfiedor** [comment]
I won't be adding any new changes to this PR,

**Pull request merged**